### PR TITLE
feat: add GOCOVERDIR for deployment

### PR DESCRIPTION
**Pull request by FrankYang0529**
https://github.com/longhorn/longhorn/issues/5681

### CODE REVIEW

File: types/deploy.go [hunk 1]
```diff
@@ -11,3 +18,69 @@ const (
 	CSISnapshotterName = "csi-snapshotter"
 	CSIPluginName      = "longhorn-csi-plugin"
 )
+
+func AddGoCoverDirToPod(pod *corev1.Pod) {
```
Comment (innobead) [inline types/deploy.go:22]: nit: add a doc comment.

File: controller/backing_image_manager_controller.go [hunk 2]
```diff
@@ -789,6 +789,8 @@ func (c *BackingImageManagerController) createBackingImageManagerPod(bim *longho
 	if err != nil {
 		return err
 	}
+	types.AddGoCoverDirToPod(podManifest)
```
Comment (innobead) [inline controller/backing_image_manager_controller.go:792]: Suggest moving this to `generateBackingImageManagerPodManifest` consistent with other changes here.

Comment (FrankYang0529) [reply controller/backing_image_manager_controller.go]: Good catch. Updated it. Thank you.

File: types/deploy.go (hunk 1 above)
Comment (FrankYang0529) [reply types/deploy.go:24]: Added doc comment for `AddGoCoverDirToPod` and `AddGoCoverDirToDaemonSet`.

**Review by innobead — APPROVED** [review]

**Pull request merged**